xds/src/main/java/io/grpc/xds/RbacFilter.java

```diff
@@ -0,0 +1,308 @@
+    GrpcAuthorizationEngine.Action authAction;
+    switch (rbacConfig.getAction()) {
+      case ALLOW:
+        authAction = GrpcAuthorizationEngine.Action.ALLOW;
+        break;
+      case DENY:
+        authAction = GrpcAuthorizationEngine.Action.DENY;
+        break;
+      case LOG:
+      default:
+        return null;
+    }
+    Map<String, Policy> policyMap = rbacConfig.getPoliciesMap();
+    List<GrpcAuthorizationEngine.PolicyMatcher> policyMatchers = new ArrayList<>();
+    for (Map.Entry<String, Policy> entry: policyMap.entrySet()) {
+      try {
+        Policy policy = entry.getValue();
+        if (policy.hasCondition() || policy.hasCheckedCondition()) {
+          return ConfigOrError.fromError(
+                  "Policy.condition and Policy.checked_condition must not set.");
+        }
+        policyMatchers.add(new PolicyMatcher(entry.getKey(),
+                parsePermissionList(policy.getPermissionsList()),
+                parsePrincipalList(policy.getPrincipalsList())));
+      } catch (Exception e) {
+        logger.log(Level.FINE, "Unexpected error parsing policy {0}", new Object[]{entry});
+        return ConfigOrError.fromError("Encountered error parsing policy: " + e);
+      }
+    }
+    return ConfigOrError.fromConfig(RbacConfig.create(new AuthConfig(policyMatchers, authAction)));
+  }
+
+  @Override
+  public ConfigOrError<? extends FilterConfig> parseFilterConfigOverride(Message rawProtoMessage) {
+    return parseFilterConfig(rawProtoMessage);
+  }
+
+  @Override
+  public ServerInterceptor buildServerInterceptor(FilterConfig config,
+                                                  @Nullable FilterConfig overrideConfig) {
+    checkNotNull(config, "config");
+    if (overrideConfig != null) {
+      config = overrideConfig;
+    }
+    AuthConfig authConfig = ((RbacConfig) config).authConfig();
+    return generateAuthorizationInterceptor(authConfig);
+  }
+
+  private ServerInterceptor generateAuthorizationInterceptor(AuthConfig config) {
+    final GrpcAuthorizationEngine authEngine = new GrpcAuthorizationEngine(config);
+    return new ServerInterceptor() {
+        @Override
+        public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+                final ServerCall<ReqT, RespT> call,
+                final Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+          AuthDecision authResult = authEngine.evaluate(headers, call);
+          logger.log(Level.FINE,
+                  "Authorization result for serverCall {0}: {1}, matching policy: {2}.",
+                  new Object[]{call, authResult.decision(), authResult.matchingPolicyName()});
+          if (GrpcAuthorizationEngine.Action.DENY.equals(authResult.decision())) {
+            Status status = Status.UNAUTHENTICATED.withDescription(
+                    "Access Denied, matching policy: " + authResult.matchingPolicyName());
+            call.close(status, new Metadata());
+            return new NoopListener<>();
```

Comment:
Just use a static (constant) instance instead of creating a new one every time.

Reply:
ServerCall.Listener is generic type, so static instance will have to bypass the unchecked warning.

Reply:
LOL, okay, I didn't consider that. You could still use create new instance each time if you don't like the casting, not a big deal. But, at least I think you don't need to define a `NoopListener` class. `new ServerCall.Listener<ReqT>() { };` is as simple as is.

Reply:
Right, it could be anonymous but static instance should be good as well.

Reply:
I mean no matter you choose to create a new instance every time or use an singleton, you don't need the `NoopListener ` class, just do the anonymous instantion.